src/tshare/participant.rs

```diff
@@ -544,74 +565,47 @@ impl TshareParticipant {
         self.local_storage
             .store::<storage::GlobalRid>(self.id(), global_rid);
 
-        let decom = self
-            .local_storage
-            .retrieve::<storage::Decommit>(self.id())?;
-
         let transcript = schnorr_proof_transcript(self.sid(), &global_rid, self.id())?;
 
-        // Generate proofs for each share.
-        let precoms = self
-            .local_storage
-            .retrieve::<storage::VecSchnorrPrecom>(self.id())?;
-
         let private_coeffs = self
             .local_storage
             .retrieve::<storage::PrivateCoeffs>(self.id())?;
 
-        let mut proofs: Vec<PiSchProof> = vec![];
-        for i in 0..precoms.len() {
-            let pk = &decom.coeff_publics[i];
-            let input = CommonInput::new(pk);
-            let precom = &precoms[i];
-            let sk = &private_coeffs[i];
-
-            let proof = PiSchProof::prove_from_precommit(
-                &self.retrieve_context(),
-                precom,
-                &input,
-                &ProverSecret::new(&scalar_to_bn(sk.as_ref())),
-                &transcript,
-            )?;
-
-            proofs.push(proof);
+        let my_private_share = Self::eval_private_share(private_coeffs, self.id());
+        let my_contant_term = Self::eval_private_share_at_zero(private_coeffs);
+        if let Some(private) = self.input.share() {
+            assert_eq!(my_contant_term, private.x);
         }
+        let public_share = EvalPublic::new(my_private_share.public_point());
```

Comment:
The `sk` used in the schnorr proof should be `my_private_share` + [all other private shares from other parties]

Parties should be able to compute the `pk` for checking every other party by using the Feldman's VSS values from the first round (another way to say this is by using the public keys they have for each party at the end of the protocol)